modules/serverless-ossm-enable-sidecar-injection-with-kourier.adoc

```diff
@@ -1,42 +1,36 @@
-// Module included in the following assemblies:
-// * serverless/networking/serverless-ossm.adoc
-// * serverless/networking/serverless-ossm-jwt.adoc
-
-[id="serverless-enable-sidecar_{context}"]
+[id="serverless-ossm-enable-sidecar-injection-with-kourier_{context}"]
 = Enabling sidecar injection for a Knative service
 
-You can add the `sidecar.istio.io/inject="true"` annotation to a Knative service to enable sidecar injection for that service.
+You can add the `sidecar.istio.io/inject="true"` annotation to a `Service` resource to enable sidecar injection for a Knative service.
 
 [IMPORTANT]
 ====
-Adding sidecar injection to pods in system namespaces such as `knative-serving` and `knative-serving-ingress` is not supported.
+Adding sidecar injection to pods in system namespaces such as `knative-serving` and `knative-serving-ingress` is not supported when Kourier is enabled.
+
+If you require sidecar injection for pods in these namespaces, see the {ServerlessProductName} documentation on _Integrating Service Mesh and OpenShift Serverless without Kourier_.
 ====
+// Add an xref here once it's enabled for modules
 
 .Procedure
 
-. Add the `sidecar.istio.io/inject="true"` annotation to the `Service` resource YAML file:
+. Add the `sidecar.istio.io/inject="true"` annotation to the `Service` resource:
 +
-.Example service YAML
+.Example service
 [source,yaml]
 ----
 apiVersion: serving.knative.dev/v1
 kind: Service
 metadata:
-  name: hello-example-1
+  name: <service_name>
 spec:
   template:
     metadata:
       annotations:
         sidecar.istio.io/inject: "true" <1>
         sidecar.istio.io/rewriteAppHTTPProbers: "true" <2>
-    spec:
-      containers:
-      - image: docker.io/openshift/hello-openshift
-        name: container
```

Comment:
Because we are only trying to illustrate in this example where the annotations for sidecar injection need to be added - the idea is not for this to be a full example of a working service that someone can copy/paste and use, but to show where they would add these annotations in their own service. This part with `containers` as far as I know is not related to sidecar injection?

Reply:
The same would be for metadata, kind, and apiVersion. Those indicate a valid resource, but then you chop it prematurely. I find it confusing, as there is no indication that is a part of the resource only.

So, either remote the metadata, leaving only the annotations path like:

```yaml
spec:
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "true" <1>
        sidecar.istio.io/rewriteAppHTTPProbers: "true" <2>
```

Or better, in my opinion, restore the image, to have minimal viable resource:

```yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: <service_name>
spec:
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "true" <1>
        sidecar.istio.io/rewriteAppHTTPProbers: "true" <2>
    spec:
      containers:
      - image: <container_image>
```

Reply:
I disagree. there can be multiple spec fields in a resource, as you show above with the containers bit. Having the metadata shows _where_ the spec field is that this needs to be added to. Otherwise if we remove them, there's no context for where this spec field goes, and someone could easily add it to the wrong place.

I will add the `...` to show that this is a truncated example, fair?

Reply:
Adding `...` is a good idea. Use the following to keep a valid YAML:

```yaml
# [...]
```

Reply:
In the OCP docs [guidelines](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output) we are specifically instructed only to use `...`
"Do not use `[…​]`, `<snip>`, or any other variant."
So I will keep it as `...`
It shouldn't matter because copying this directly would not work anyway so the YAML doesn't need to be valid, the examples contain variables etc and are not supposed to be copy/pasted anyway

Reply:
Okay. Cool. SMdocs are having examples without `metadata` field - That's also preventing the dump copy-paste. JFYI